### PR TITLE
docs: Update review documentation

### PR DESCRIPTION
**Pull request by sils**
### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [ ] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### CODE REVIEW

File: docs/Developers/Review.rst [hunk 1]
```diff
@@ -3,19 +3,30 @@ Reviewing
 
 This document is a guide to coala's review process.
 
+Am I Good Enough to Do Code Review?
+-----------------------------------
+
+**Yes!** If you already fixed a newcomer issue.
```
Comment (Adrianzatreanu) [inline docs/Developers/Review.rst]: i dont like this phrase as it doesnt even have a verb

File: docs/Developers/Review.rst [hunk 2]
```diff
@@ -3,19 +3,30 @@ Reviewing
...
+
+Please review at least one Pull Request for every review you receive. Start
+with other people's newcomer issues and advance as you do with your own issues.
+
+You can and should help us review source code. Especially try to review source
+code of others and share what you have learnt with them. You can use acks and
+unacks like everyone else and ``cobot`` allows you to set PRs to WIP even.
+
+Try to be thorough: generally, if you don't find an issue with a pull request,
```
Comment (Adrianzatreanu) [inline docs/Developers/Review.rst]: i dont understand this paragraph at first glance => a) it is badly stated or b) i havent got enough knowledge of what youre saying here

File: docs/Developers/Review.rst [hunk 3]
```diff
@@ -24,36 +35,33 @@ The review process for coala is as follows:
...
 
-5. The developers will acknowledge the commits by writing
-
-    - ``ack commit_SHA``, in case the commit is ready, or
-
-    - ``unack commit_SHA / commit_SHA needs work`` in case it is not ready yet
-      and needs some more work.
-
-6. If the commits are not linearly mergeable into master, rebase and go
+   If you don't have write access to coala, you can change the labelling using
```
Comment (Adrianzatreanu) [inline docs/Developers/Review.rst]: change the labels, not the labelling. we dont mean the action here

File: docs/Developers/Review.rst [hunk 4]
```diff
@@ -24,36 +35,33 @@ The review process for coala is as follows:
...
+   ``cobot pull wip <URL>`` or ``cobot pull needreview <URL>``.
```
Comment (Adrianzatreanu) [inline docs/Developers/Review.rst]: i think "cobot mark wip <URL>" is better, dont you think?

Comment (sils) [inline docs/Developers/Review.rst]: hm not sure, could work.

Comment (sils) [inline docs/Developers/Review.rst]: cobot mark pending review with a space?

File: docs/Developers/Review.rst [hunk 5]
```diff
@@ -72,11 +80,13 @@ For the Reviewers
 -  Check the commit message.
 -  Take a look at continuous integration results in the end even if they
    pass.
--  Coverage shall not fall.
--  Scrutinizer often yields helpful results.
+-  Coverage must not fall.
+-  Be sure to assure that the tests cover all corner cases and validate the
+   behaviour well. E.g. for bear tests just testing for a good and bad file
+   is *not* sufficient.
```
Comment (Adrianzatreanu) [inline docs/Developers/Review.rst]: i think "not" should be bolded, not italic-ed

File: docs/Developers/Review.rst (hunk 4 above)
Comment (Adrianzatreanu) [inline docs/Developers/Review.rst]: sounds good

**Comment by sils** [comment]
related: https://gitlab.com/coala/cobot/merge_requests/39

File: docs/Developers/Review.rst [hunk 6]
```diff
@@ -3,19 +3,32 @@ Reviewing
...
+-----------------------------------
+
+**Yes,** if you already fixed a newcomer issue.
+
+Please review at least one Pull Request for every review you receive. Start
+with other people's newcomer issues and advance as you do with your own issues.
+
+You can and should help us review source code. Especially try to review source
+code of others and share what you have learnt with them. You can use acks and
+unacks like everyone else and ``cobot`` allows you to set PRs to WIP even.
```
Comment (Mixih) [inline docs/Developers/Review.rst]: allows you to set PRs to WIP even -> even allows you to set PRs to WIP

File: docs/Developers/Review.rst [hunk 7]
```diff
@@ -24,36 +37,33 @@ The review process for coala is as follows:
...
-following apply:
-
--  The build/tests pass on all services. (circleci, appveyor)
--  Scrutinizer shows passed. (That is: no new issues, no new classes
-   with rating D or worse, project quality metric may only get better.)
--  All statements and branches are covered by your tests. (codecov.io)
-
-The coverage values may go down by a commit, however this is to be
-avoided. Tests must work for every commit.
+It is only allowed to merge a pull request into master if all ``required``
```
Comment (Mixih) [inline docs/Developers/Review.rst]: why is "required" in code formatting?

File: docs/Developers/Review.rst [hunk 8]
```diff
@@ -3,19 +3,32 @@ Reviewing
 
 This document is a guide to coala's review process.
 
+Am I Good Enough to Do Code Review?
+-----------------------------------
+
+**Yes,** if you already fixed a newcomer issue.
+
+Please review at least one Pull Request for every review you receive. Start
+with other people's newcomer issues and advance as you do with your own issues.
```
Comment (Mixih) [inline docs/Developers/Review.rst]: "advance as you do with your own issues" is a little confusing and could use some rewording imo

File: docs/Developers/Review.rst (hunk 7 above)
Comment (sils) [inline docs/Developers/Review.rst]: it's this github "required" label, wanted to highlight it somehow

Comment (Mixih) [inline docs/Developers/Review.rst]: ah ok

**Comment by sils** [comment]
CC @sims1253 how is this?

File: docs/Developers/Review.rst [hunk 9]
```diff
@@ -3,19 +3,51 @@ Reviewing
 
 This document is a guide to coala's review process.
 
+Am I Good Enough to Do Code Review?
+-----------------------------------
+
+**Yes,** if you already fixed a newcomer issue.
+
+But be careful: **bad reviews will scare away other contributors.**
```
Comment (fneu) [inline docs/Developers/Review.rst]: I find this a little harsh. Especially as one out of 3 bolded statements.

Comment (fneu) [inline docs/Developers/Review.rst]: Since the first and following paragraph seem to foremost encourage reviewing, maybe remove this here and add it further down where you talk about being transparent and polite (l. 26).

File: docs/Developers/Review.rst [hunk 10]
```diff
@@ -3,19 +3,51 @@ Reviewing
...
+Generally follow this process:
+
+1. Check if the change is helping us. Sometimes people propose changes that are
+   only helpful for specific usecases but may break others. The concept has to
+   be good. Consider engaging in a discussion on gitter if you are unsure!
+2. Check for automatic builds. Give the contributor hints on how he can resolve
+   them.
+3. Review the actual code. Suggest improvements and simplifications.
+
+**Be sure to not value quantity over quality! Be transparent and polite:
```
Comment (fneu) [inline docs/Developers/Review.rst]: Bolding two statements is like writing if clauses in one line. Not as helpful and imidiately obvious as it should be. Consider to break these sentences up a little bit, maybe also adding the "try not to scare people away thing" somewhere around here.

File: docs/Developers/Review.rst [hunk 11]
```diff
@@ -3,19 +3,51 @@ Reviewing
 
 This document is a guide to coala's review process.
 
+Am I Good Enough to Do Code Review?
+-----------------------------------
+
+**Yes,** if you already fixed a newcomer issue.
```
Comment (fneu) [inline docs/Developers/Review.rst]: One thing I'm generally missing is how helpful it can be for newcomers to engage in reviewing. They are not only doing something for us, they are also helping themselves. They learn about how to improve their own commits, different internals of coala, different people, how to code better in general, ...

Comment (sils) [inline docs/Developers/Review.rst]: @fneu mind phrasing a pragraph?

**Comment by sims1253** [comment]
ack bc9114df59aee0c70676ee7628615d1d745a27eb

File: docs/Developers/Review.rst [hunk 12]
```diff
@@ -3,19 +3,55 @@ Reviewing
...
+
+Reviewing can help you understand the other side of the table and learn more
+about coala and python. When reviewing you will get to know new people, more
+code and it ultimately helps you to become a better coder than you could do
+on your own.
+
+You can totally help us review source code. Especially try to review source
+code of others and share what you have learnt with them. You can use acks and
+unacks like everyone else and ``cobot`` even allows you to set PRs to WIP.
```
Comment (sims1253) [inline docs/Developers/Review.rst]: maybe link to the section below?

File: docs/Developers/Review.rst [hunk 13]
```diff
@@ -24,36 +60,33 @@ The review process for coala is as follows:
...
+      yet and needs some more work or
+    - ``reack commit_SHA`` in case the commit was acknowledged before, was
+      rebased without conflicts and the rebase did not introduce logical
+      problems.
+4. If the commits are not linearly mergeable into master, rebase and go
    to step one.
-7. All commits are acknowledged and fit linearly onto master. All
+5. All commits are acknowledged and fit linearly onto master. All
    continuous integration services (as described below) pass. Anyone
    with collaborator permission may leave the ``@rultor merge`` command
```
Comment (sims1253) [inline docs/Developers/Review.rst]: maybe tell them that a maintainer will do this? or we might get newcomers merging stuff

**Comment by sims1253** [comment]
ack fb71a9d

**Comment by sims1253** [comment]
ack cb26a05

**Comment by sims1253** [comment]
@rultor merge
